test/catch_main/CMakeLists.txt

```diff
@@ -30,6 +30,10 @@ set_target_properties(CatchMain PROPERTIES
 )
 
 target_compile_definitions(CatchMain PUBLIC "CATCH_CONFIG_FAST_COMPILE")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "PGI")
+    # Workaround for STL atomic issue: https://forums.developer.nvidia.com/t/support-for-atomic-in-libstdc-missing/135403/2
+    target_compile_definitions(CatchMain PUBLIC "__GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1")
```

Comment:
Why do we need both? Shouldn't it be enough to set it either with the build system OR within the code?

Reply:
I set it in the code so it works for the Alpaka stand-alone headers.

The flag in the build system only affects the tests, which is required because catch2 itself also includes <atomic> and I did not want to touch the catch2 code.

Reply:
An alternative would be to put it into the Alpaka target and the stand-alone headers instead of where `<atomic>` is included in the code.

Reply:
I changed it just add the thing to the alpaka target interface and the CatchMain target. Any other Program using alpaka is likely to include `<atomic>` somewhere else, maybe indirectly through a different library, thus fixing it in the stand-alone headers does not seem very useful after all.